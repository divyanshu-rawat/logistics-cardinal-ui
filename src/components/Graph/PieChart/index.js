import React from 'react';
import PropTypes from 'prop-types';
import { PieChart as RCPieChart } from 'recharts';

function PieChart({ children, ...props }) {
  return <RCPieChart {...props}>{children}</RCPieChart>;
}

PieChart.propTypes = {
  /**
   * Components placed inside the PieChart
   */
  children: PropTypes.node.isRequired,
};

export default PieChart;
