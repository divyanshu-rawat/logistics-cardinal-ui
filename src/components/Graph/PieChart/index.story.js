import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { Cell } from 'recharts';

import withTests from '../../../utils/withTests';
import { GRAPH } from '../../../../.storybook/sections';

import Pie from '../Pie';
import PieChart from '.';
import { SIZES } from '../../utils/constants';

storiesOf(`${GRAPH}PieChart`, module)
  .addDecorator(withTests('PieChart/tests/index'))
  .add(
    'Simple',
    withInfo()(() => (
      <PieChart width={400} height={400}>
        <Pie
          data={[{ value: 30 }, { value: 20 }, { value: 15 }]}
          dataKey="value"
          innerRadiusSize={SIZES.xs}
          outerRadiusSize={SIZES.lg}
        >
          <Cell key="CellOne" fill="rgba(79, 127, 177, 0.3)" stroke="#4f7fb1" />
          <Cell key="CellTwo" fill="rgba(109,179,36,0.3)" stroke="#6DB324" />
          <Cell key="CellThree" fill="#662147" stroke="#E6499F" />
        </Pie>
      </PieChart>
    )),
  );
