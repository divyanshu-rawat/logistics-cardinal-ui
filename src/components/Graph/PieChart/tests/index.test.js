import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PieChart from '..';

const renderedComponent = () => shallow(<PieChart>Awesome chart</PieChart>);

describe('<PieChart />', () => {
  it('should render the component', () => {
    const tree = toJson(renderedComponent());
    expect(tree).toMatchSnapshot();
  });
});
