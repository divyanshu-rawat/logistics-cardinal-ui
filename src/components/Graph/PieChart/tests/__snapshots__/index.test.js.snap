// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PieChart /> should render the component 1`] = `
<PieChart
  barCategoryGap="10%"
  barGap={4}
  cx="50%"
  cy="50%"
  endAngle={360}
  innerRadius={0}
  layout="centric"
  margin={
    Object {
      "bottom": 5,
      "left": 5,
      "right": 5,
      "top": 5,
    }
  }
  outerRadius="80%"
  reverseStackOrder={false}
  stackOffset="none"
  startAngle={0}
>
  Awesome chart
</PieChart>
`;
